### ISSUE TITLE
inherited methods do not work

### ISSUE THREAD

**Issue by pdjohe** [astefanutti/metrics-cdi#15]
Consider the following:

Class A:

```
public abstract class A {
    public void test2() {}
}
```

Class B:

```
@Timed
@ApplicationScoped
public class B extends A {
    public void test1() {}
}
```

Class C:

```
public class C {
    @Inject
    private B b;
    public void runTest1() {
        b.test1();
    }
    public void runTest2() {
        b.test2();
    }
}
```

runTest1 will work and be timed
runTest2 will fail:

```
java.lang.UnsupportedOperationException
    at io.astefanutti.metrics.cdi.MetricResolver$DoesNotHaveMetric.metricName(MetricResolver.java:199) ~[metrics-cdi-1.3.6.jar:1.3.6]
    at io.astefanutti.metrics.cdi.TimedInterceptor.timedCallable(TimedInterceptor.java:63) ~[metrics-cdi-1.3.6.jar:1.3.6]
    at io.astefanutti.metrics.cdi.TimedInterceptor.timedMethod(TimedInterceptor.java:54) ~[metrics-cdi-1.3.6.jar:1.3.6]
```

This is because it is assumed that the method.getDeclaringClass() is the actual class that is annotated. That is not the case with inheritance. The class should not be pulled from the Method object but from:

```
Class<?> proxiedClass = invocationContext.getTarget().getClass();
```

Note also that this should probably work as well:

Class A:

```
@Timed
public abstract class A {
    public void test2() {}
}
```

Class B:

```
@ApplicationScoped
public class B extends A {
    public void test1() {}
}
```

such that both test1() and test2() are both timed.

**Comment by astefanutti** [comment]
@pdjohe thanks for the report. I agree with your analysis.

I'll work on it ASAP. Otherwise feel free to contribute a PR and I'll be happy to merge it and cut a release for you.

**Comment by astefanutti** [comment]
Fixed in e3afc46398a8f613fbf841984a64a838d7b7898e.
